src/ModelContextProtocol.Core/Protocol/ElicitRequestParams.cs

```diff
@@ -126,7 +126,19 @@ public class Converter : JsonConverter<PrimitiveSchemaDefinition>
                     switch (propertyName)
                     {
                         case "type":
-                            type = reader.GetString();
+                            if (reader.TokenType == JsonTokenType.String)
+                            {
+                                type = reader.GetString();
+                            }
+                            else if (reader.TokenType == JsonTokenType.StartArray)
+                            {
+                                var types = JsonSerializer.Deserialize(ref reader, McpJsonUtilities.JsonContext.Default.StringArray);
+                                if (types is [var nullableType, "null"])
```

Comment:
What if the keyword uses `["null", "object"]`, an equally valid form?

Reply:
@eiriktsarpalis Done.